tests/unit/test_spaces.cpp

```diff
@@ -305,6 +305,7 @@ TEST_F(SpacesTest, l2_8_double) {
     switch (optimization) {
     case ARCH_OPT_AVX512: // TODO: add comparison when AVX and AVX512 is implemented
     case ARCH_OPT_AVX:
+        ASSERT_EQ(baseline, FP64_L2SqrSIMD8Ext_AVX(v, v, dim));
```

Comment:
We should ask ourselves how did these tests pass even though you used (mistakenly, I think) `FP32` functions instead of `FP64`... If so, it is better to use a vector in the tests whose representation is different in `double` and `float` is different, so that such errors will be caught.